maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultPluginValidationManager.java

```diff
@@ -247,7 +247,11 @@ private String pluginOccurrence(MavenSession mavenSession) {
         File currentPom = prj.getFile();
         if (currentPom != null) {
             File rootBasedir = mavenSession.getTopLevelProject().getBasedir();
-            result += " (" + rootBasedir.toPath().relativize(currentPom.toPath()) + ")";
+            if (currentPom.getPath().startsWith(rootBasedir.getPath())) {
```

Comment:
I wonder if the path should be absoluted / normalized.  Also, this should be done against the new `Session#getRootDirectory()` imho:
```
            Path root = mavenSession.getRootDirectory();
            Path current = currentPom.toPath().toAbsolutePath().normalize();
            if (current.startsWith(root)) {
                current = root.relativize(current);
            }
            result += " (" + current + ")";
```

Reply:
Note that `rootDirectory` always refer to the root of the project, while `topDirectory` refer to the current path (those may be different when building a subset of a project when using `cd xxx && mvn ...` or `mvn -f xxx ...`

Reply:
`rootDirectory` seems to be available for maven 4 only...

Reply:
On 3.9.x, you could use `session.getRequest().getMultiModuleProjectDirectory()`.

@cstamas what's your take on that ?  would you rather keep poms relative to the top directory ?

Reply:
I've refactored the parts using your ideas (determine the relative path first before adding to the result string). The commits should probably be squashed before merged.

I've changed it also for `pluginDeclaration()`, to have consistent behavior.

Open question is now: Should I use `session.getRequest().getMultiModuleProjectDirectory().toPath()` instead of `mavenSession.getTopLevelProject().getBasedir().toPath()` for the root directory (for both pluginDeclaration() and pluginOccurrence()?

Reply:
I think we need to be consistent with
https://github.com/apache/maven/blob/e9d570889ffce29c3f92494510499ae0e32a02cf/maven-embedder/src/main/java/org/apache/maven/cli/event/ExecutionEventLogger.java#L299-L300
So better continue using the top directory.  However, does the ExecutionEventLogger suffer from the same problem ?

Reply:
> And can we make the local variable name consistent, i.e. name it `topBasedir` instead of `rootBasedir` ?

Done.

I've also updated ExecutionEventLogger. It seems, this was the only other place, where Path::relativize is used.